frontend/app/src/next/components/members/members-table.tsx

```diff
@@ -0,0 +1,197 @@
+                      Remove
+                    </Button>
+                  )}
+                </TableCell>
+              </TableRow>
+            ))}
+          </TableBody>
+        </Table>
+      </div>
+      {invites && invites.length > 0 && (
+        <>
+          <Separator className="my-8" />
+
+          <h3 className="text-xl font-semibold leading-tight text-foreground mb-4">
+            Pending Invitations
+          </h3>
+          <InvitesTable
+            data={invites}
+            isLoading={isLoadingInvites}
+            onRevokeClick={(invite) => {
+              setRevokeInvite(invite);
+            }}
+            emptyState={
+              <div className="flex flex-col items-center justify-center gap-4 py-8">
+                <p className="text-sm text-muted-foreground">
+                  No pending invitations.
+                </p>
+              </div>
+            }
+          />
+        </>
+      )}
+
+      {removeMember && (
+        <RemoveMemberForm
+          member={removeMember}
+          close={() => {
+            setRemoveMember(null);
+            refetch();
+          }}
+        />
+      )}
+
+      {revokeInvite && (
+        <RevokeInviteForm
+          invite={revokeInvite}
+          close={() => {
+            setRevokeInvite(null);
+          }}
+        />
+      )}
+    </>
+  );
+}
+
+function MembersTableSkeleton() {
+  return (
+    <div className="rounded-md border">
+      <Table>
+        <TableHeader>
+          <TableRow>
+            <TableHead>User</TableHead>
+            <TableHead>Email</TableHead>
+            <TableHead>Role</TableHead>
+            <TableHead>Joined</TableHead>
+            <TableHead className="w-[100px] text-right"></TableHead>
+          </TableRow>
+        </TableHeader>
+        <TableBody>
+          {Array.from({ length: 5 }).map((_, i) => (
+            <TableRow key={i}>
+              <TableCell>
+                <Skeleton className="h-5 w-[150px]" />
+              </TableCell>
+              <TableCell>
+                <Skeleton className="h-5 w-[200px]" />
+              </TableCell>
+              <TableCell>
+                <Skeleton className="h-5 w-[100px]" />
+              </TableCell>
+              <TableCell>
+                <Skeleton className="h-5 w-[120px]" />
+              </TableCell>
+              <TableCell className="text-right">
+                <Skeleton className="h-8 w-[80px] ml-auto" />
+              </TableCell>
+            </TableRow>
+          ))}
+        </TableBody>
+      </Table>
+    </div>
+  );
+}
+
+function formatDate(date?: string) {
+  if (!date) {
+    return '-';
+  }
+  return new Date(date).toLocaleDateString();
+}
+
+function formatRole(role: TenantMemberRole) {
+  switch (role) {
+    case TenantMemberRole.ADMIN:
+      return 'Admin';
+    case TenantMemberRole.MEMBER:
+      return 'Member';
+    case TenantMemberRole.OWNER:
+      return 'Owner';
+    default:
+      return role;
```

Comment:
I've been really liking adding exhaustiveness checks for things like this so the compiler will yell if a new enum value gets added:

```typescript
default:
  const exhaustivenessCheck: never = role;
  throw new Error("unhandled role")
```

(we need to disable an eslint rule for this, but it's a nice pattern for maintainability going forward imo